pandas/core/strings/object_array.py

```diff
@@ -308,21 +308,38 @@ def f(x):
 
         return self._str_map(f)
 
-    def _str_split(self, pat=None, n=-1, expand=False):
+    def _str_split(
+        self,
+        pat: str | re.Pattern | None = None,
+        n=-1,
+        expand=False,
+        regex: bool | None = None,
+    ):
```

Comment:
> while str_split does not. str_split handles the case when regex=None by using the weird logic with len(pat),

then shouldn't we ad `case` and `flags` and make these consistent? what do we need to deprecate here?

yes there is weird logic by using `len(pat)` but this is *new* logic you are adding no?

Reply:
I did not add the len(pat) logic - it has always been there. I purposely didn't cut it out to maintain current behavior. I think that we should remove it in the future. 
https://github.com/pandas-dev/pandas/blob/2fa2d5cc5c16fcf546cd641e78dd270b66bef5f7/pandas/core/strings/object_array.py#L317-L325
In this PR, I am simply adding the regex flag, as requested by several issues. I can go ahead and add the `case` and `flags` if you think that is a good idea. 
Thanks

Reply:
I saw that you didn't change the current `len(pat)` logic, which I don't like either. Maybe it is better if we are adding an explicit `regex` argument to permit only True, False, and remove the `len(pat)` logic, which only confuses things.
Of course need a release note and a versionchanged note.

Reply:
I'm happy to get rid of the len(pat) logic and thus the `regex=None` option, but I didn't want to introduce a breaking change. 

If you guys want me to go ahead, what should the `regex` default be? I don't want to break anyone's existing code. For example, if we make `regex` default to `True`, it will break someone's code splitting on a period (`pat="."`)

Reply:
@jreback what do you think on breaking change here. I would be inclined to set the default to `True`, since it uses Regex in every case where the pattern length is is greater than 1, and in most cases where the pattern length is 1, then regex would give the same result as string anyway, so it might only break in a few cases.

Reply:
While I agree that removing this logic would be great, I'd be worried about changing the default to `True` without a deprecation. While most cases won't be affected like you mention, I'd imagine many users (who may not even know what a regex is) often split on punctuation like "." or "-".  Many of the issues referenced here (or others for `replace`) stem from the user not realizing that the pattern is being treated as regex - I'd guess we'd see a bunch more of those if the default is changed

Reply:
since this is a new method we could change things, but yeah maybe this is too much for now.

ok what i think we should do is this.

factor to a common method and use it here. when we deprecate this it will deprecate in both places (yes even though this is a new method it is fine). its at least consistent.